### PR TITLE
More efficient pool with pointers

### PR DESCRIPTION
**Pull request by myleshorton**
Technically should profile this, but I think this should be a significant improvement in pool performance.

This is based on the discussion here: https://blog.mike.norgate.xyz/unlocking-go-slice-performance-navigating-sync-pool-for-enhanced-efficiency-7cb63b0b453e

I'll add some benchmarks.

### CODE REVIEW

**Comment by myleshorton** [comment]
OK here's the benchmark with the code in this PR:

```
$ go test -run=XXX -bench BenchmarkPool
goos: darwin
goarch: arm64
pkg: github.com/getlantern/ipproxy
BenchmarkPool-8   	145936260	         8.130 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/getlantern/ipproxy	2.151s
```

**Comment by myleshorton** [comment]
Here's with the current code on master:

```
$ go test -run=XXX -bench BenchmarkPool
goos: darwin
goarch: arm64
pkg: github.com/getlantern/ipproxy
BenchmarkPool-8   	49513612	        25.05 ns/op	      24 B/op	       1 allocs/op
PASS
ok  	github.com/getlantern/ipproxy	2.249s
```

File: pool.go [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 
 // Buffer pool for forwarding UDP packets
-var udpBufPool = sync.Pool {
-    New: func() any {
-        return make([]byte, maxUDPPacketSize)
-    },
+var udpBufPool = sync.Pool{
+	New: func() any {
+		b := make([]byte, maxUDPPacketSize)
+		return &b
```
Comment (atavism) [inline pool.go:13]: Nice.. makes sense we can avoid an extra allocation here returning a pointer type

**Comment by atavism** [comment]
LGTM. Thanks, @myleshorton!

**Pull request merged**